indexer/feature_impl.cpp

```diff
@@ -65,12 +65,12 @@ bool IsHouseNumberDeepCheck(strings::UniString const & s)
 
 bool IsHouseNumber(string const & s)
 {
-  return (!s.empty() && IsDigit(s[0]));
+  return (!s.empty() && (IsDigit(s.front()) || IsDigit(s.back())));
 }
 
 bool IsHouseNumber(strings::UniString const & s)
 {
-  return (!s.empty() && IsDigit(s[0]));
+  return (!s.empty() && (IsDigit(s.front()) || IsDigit(s.back())));
```

Comment:
It would be slower. Moreover, I'm afraid that this will dramatically decrease performance of our search. Previously during search we tried to consume house number from the first token that looks like a number, i.e. we try to find a building that matches to all subsequences of tokens that start with a token with the first digit. Now all possible subsequences of tokens that starts or ends with a digit will be recognized as a house number.